backend-service/src/main/java/com/smartparkinglot/backendservice/exceptions/ExceptionHandler.java

```diff
@@ -4,51 +4,58 @@
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 
 import java.util.ArrayList;
 import java.util.List;
 
-@ControllerAdvice
+@RestControllerAdvice
 public class ExceptionHandler extends ResponseEntityExceptionHandler {
 
     @org.springframework.web.bind.annotation.ExceptionHandler(Exception.class)
-    public final ResponseEntity<Object> handleAllExceptions(Exception ex) {
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    public final ResponseEntity<ErrorResponse> handleAllExceptions(Exception ex) {
         List<String> details = new ArrayList<>();
         details.add(ex.getLocalizedMessage());
         ErrorResponse err = new ErrorResponse("Server Error", details);
-        return new ResponseEntity(err, HttpStatus.INTERNAL_SERVER_ERROR);
+        return new ResponseEntity<ErrorResponse>(err, HttpStatus.INTERNAL_SERVER_ERROR);
     }
 
-    @org.springframework.web.bind.annotation.ExceptionHandler(AccountActivatedException.class)
-    public final ResponseEntity<Object> handleAccountActivatedExceptions(Exception ex) {
+    @org.springframework.web.bind.annotation.ExceptionHandler(AccountDeactivatedException.class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    public final ResponseEntity<ErrorResponse> handleAccountActivatedExceptions(Exception ex) {
         List<String> details = new ArrayList<>();
         details.add(ex.getLocalizedMessage());
         ErrorResponse err = new ErrorResponse("INFO", details);
-        return new ResponseEntity(err, HttpStatus.ALREADY_REPORTED);
+        return new ResponseEntity<ErrorResponse>(err, HttpStatus.CONFLICT);
     }
 
     @org.springframework.web.bind.annotation.ExceptionHandler(NotFoundException.class)
-    public final ResponseEntity<Object> handleNotFoundException(Exception ex) {
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public final ResponseEntity<ErrorResponse> handleNotFoundException(Exception ex) {
         List<String> details = new ArrayList<>();
         details.add(ex.getLocalizedMessage());
         ErrorResponse error = new ErrorResponse("Not found", details);
-        return new ResponseEntity(error, HttpStatus.NOT_FOUND);
+        return new ResponseEntity<ErrorResponse>(error, HttpStatus.NOT_FOUND);
     }
 
     @org.springframework.web.bind.annotation.ExceptionHandler(AlreadyExistsException.class)
```

Comment:
Is it necessary to use `@ExceptionHandler` annotation like `@org.springframework.web.bind.annotation.ExceptionHandler` ?

Reply:
It seems so regarding to swagger configurations. Without annotations swagger doesn't implements other than 2** statuses

Reply:
Oh, you are pointing the long annotation instead of just "@ExceptionHandler". IDE does that automatically. I tried to fix that but it keep insisting on that.